### DISCUSSION TITLE
Alert provisioning: view yaml when saving

### DISCUSSION THREAD

**Discussion by holmesb** [grafana/grafana#66541] [Alerting]
Hi,

Currently gitops workflow (provisioning) for **dashboards** is:
1. User develops dash using UI and presses save.
2. Instead of saving, the user is shown json:
![image](https://user-images.githubusercontent.com/5072156/231970215-beaf9df7-dcbb-4162-a6b8-22d1e517459d.png)
which they copy into a repo, and change is deployed declaratively with all the gitops goodness: repeatability, rationale is captured, rollback is easy, peer-review, Grafana is stateless and easy to manage, etc.

This also works for legacy alerts since they were built into dashboards.

This is not possible using the new **unified alerts**. After the first alert is declared, the UI [cannot ](https://github.com/grafana/grafana/discussions/53490)be used. If the user tries to define an alert using the UI, when they save, instead of outputting the yaml, they are just shown an [error](https://github.com/grafana/grafana/pull/50835):
![image](https://user-images.githubusercontent.com/5072156/231969249-23a6ced9-b278-4011-a027-02c726fdf250.png)

So I propose this be made the same as the dashboard workflow. Instead of an error, pressing "Save" or "Save and Exit" buttons will display the yaml of that alert, which can then be placed in git. @yuri-tceretian 

Cheers.

PS. I'm aware can use the [export api](https://github.com/grafana/grafana/discussions/53490#discussioncomment-5504877) to extract the yaml of an alert, but this is hardly intuitive. Should be as easy as for dashboards.

Also this is partially related to #53490, which is asking for the ability to edit existing provisioned alerts in the UI. Whereas this request is asking for the ability to see declared alert yaml when pressing save - both for new alerts (when there are existing provisioned ones), and for existing ones (after #53490 has been delivered).

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82721. Feel free to continue the discussion around this there. Thank you!
